### PR TITLE
Feature/exclude files

### PR DESCRIPTION
**Pull request by shaharkazaz**
* Allow skipping files
* Add dry run option
* Refactor configuration options (Breaking)
* Infra changes
* Fixed playground

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -98,16 +100,89 @@ npx import-conductor -s customer.component.ts -p @myorg
...
+
+```shell script
+import-conductor --ignore mySrc/**/*some.ts main.ts
+import-conductor -i mySrc/**/*some.ts main.ts
+```
+
+**\*Note**: you can also skip a file by adding the following comment at the top:
+
+```typescript
+// import-conductor-skip
```
Comment (kreuzerk) [inline README.md:121]: 👍

File: src/conductor/get-third-party.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import pkgUp from 'pkg-up';
+
+import { parseJsonFile } from '../helpers';
 import fs from 'fs';
```
Comment (kreuzerk) [inline src/conductor/get-third-party.ts:4]: I would also try to use the same pattern of ordered imports - like fs needs to come at the beginning

File: src/conductor/optimize-imports.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
+import ts from 'typescript';
+import chalk from 'chalk';
+import { getConfig } from '../config';
+import simpleGit, { SimpleGit } from 'simple-git';
+import { readFileSync, writeFileSync } from 'fs';
```
Comment (kreuzerk) [inline src/conductor/optimize-imports.ts:5]: Import order

File: src/conductor/optimize-imports.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  skipped: 'skipped',
+  reordered: 'reordered',
+};
+
+export async function optimizeImports(filePath: string): Promise<string> {
+  if (!/\.tsx?$/.test(filePath)) {
+    return actions.none;
+  }
+
+  const fileContent = readFileSync(filePath);
```
Comment (kreuzerk) [inline src/conductor/optimize-imports.ts:25]: could we do this earlier? - like we shouldn't optimize the imports at all if the file is ignored.

**Review by kreuzerk — COMMENTED** [review]
Fantastic work @shaharkazaz!
I left a couple of nodes as improvements.

File: hooks/pre-commit.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
+const fs = require('fs');
+const path = require('path');
+const chalk = require('chalk');
+const gitChangedFiles = require('git-changed-files');
+
+gitChangedFiles({ showCommitted: false }).then(({ unCommittedFiles }) => preCommit(unCommittedFiles));
+
+function forbiddenTokens(stagedFiles) {
```
[the remark this answers is not in the document]
Comment (shaharkazaz) [reply hooks/pre-commit.js:8]: It's from a  [blog post](https://itnext.io/writing-custom-git-hooks-with-nodejs-2d53732865aa) I wrote, but maybe I will publish it as an open-source, let me think about it 👍

File: playground/sampleDir/baz/baz.ts [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import {foo} from '../foo/foo';
 import {bar} from '../bar/bar';
 
-import {Observable} from 'rxjs';
+import {Observable} from 'typescript';
```
[the remark this answers is not in the document]
Comment (shaharkazaz) [reply playground/sampleDir/baz/baz.ts:4]: Yes I agree, it's a mistake, I forgot to do it 😄

File: src/conductor/optimize-imports.ts (hunk 4 above)
Comment (shaharkazaz) [reply src/conductor/optimize-imports.ts:25]: we are not optimizing if he is ignored (we return with `actions.skipped`), I'm not sure what you mean.

**Review by kreuzerk — APPROVED** [review]

**Pull request merged**